src/OrbitGl/MainThreadExecutor.h

```diff
@@ -65,6 +67,134 @@ class MainThreadExecutor {
     return future;
   }
 
+  template <typename F>
+  [[nodiscard]] auto ScheduleAfter(const orbit_base::Future<void>& future, F&& functor)
+      -> orbit_base::Future<std::decay_t<decltype(functor())>> {
+    CHECK(future.IsValid());
+
+    using ReturnType = std::decay_t<decltype(functor())>;
+
+    // Since std::function needs to be copyable, promise also needs to be copyable.
+    auto promise = std::make_shared<orbit_base::Promise<ReturnType>>();
+    orbit_base::Future<ReturnType> resulting_future = promise->GetFuture();
+
+    // Since MSVC2017 crashes with a compiler error when using `if constexpr` inside of lambda, we
+    // have to put the check on the outside. That's unfortunate since it leads to four almost
+    // identical version of the same code.
+    if constexpr (std::is_same_v<ReturnType, void>) {
+      auto continuation = [functor = std::forward<F>(functor), executor_weak_ptr = weak_from_this(),
+                           promise = std::move(promise)]() mutable {
+        auto executor = executor_weak_ptr.lock();
+        if (executor == nullptr) return;
```

Comment:
This deserves some comment clarifying if the executer destruction can happen during orderly shutdown. Normally we would want to complete every scheduled action, and it could be argued that ScheduleAfter is also a form of scheduling.

Reply:
Good point. I added some comment which explains the function's behaviour. I wanted to do that anyway.